test/internal/kubernetes/utils.go

```diff
@@ -233,6 +233,24 @@ func WaitForPodsRunning(ctx context.Context, clientset *kubernetes.Clientset, na
 	return errors.Wrap(retrier.Do(ctx, checkPodIPsFn), "failed to check if pods were running")
 }
 
+func WaitForPodsDelete(ctx context.Context, clientset *kubernetes.Clientset, namespace, labelselector string) error {
+	podsClient := clientset.CoreV1().Pods(namespace)
+
+	checkPodsDeleted := func() error {
+		podList, err := podsClient.List(ctx, metav1.ListOptions{LabelSelector: labelselector})
+		if err != nil {
+			return errors.Wrapf(err, "could not list pods with label selector %s", labelselector)
+		}
+		if len(podList.Items) != 0 {
+			return errors.Errorf("pods still present: %+v", podList.Items)
```

Comment:
How does the output of podList.Items look?

Reply:
```
"pods still present: [{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name:win-load-test-6fdc4cbfc5-4mhp7 GenerateName:win-load-test-6fdc4cbfc5- Namespace:load-test SelfLink: UID:7a51ae9d-60b4-41c4-90c4-fbda0b8fd78c ResourceVersion:344945 Generation:0 CreationTimestamp:2023-09-27 10:25:11 -0700 PDT DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[load-test:true os:windows pod-template-hash:6fdc4cbfc5] Annotations:map[] OwnerReferences:[{APIVersion:apps/v1 Kind:ReplicaSet Name:win-load-test-6fdc4cbfc5 UID:edc399ee-5440-4992-ab05-08949af0a223 Controller:0xc0000ff92f BlockOwnerDeletion:0xc0000ff94f}] Finalizers:[] ManagedFields:[{Manager:kube-controller-manager Operation:Update APIVersion:v1 Time:2023-09-27 10:25:11 -0700 PDT FieldsType:FieldsV1 FieldsV1:{\"f:metadata\":{\"f:generateName\":{},\"f:labels\":{\".\":{},\"f:load-test\":{},\"f:os\":{},\"f:pod-template-hash\":{}},\"f:ownerReferences\":{\".\":{},\"k:{\\\"uid\\\":\\\"edc399ee-5440-4992-ab05-08949af0a223\\\"}\":{}}},\"f:spec\":{\"f:containers\":{\"k:{\\\"name\\\":\\\"noop\\\"}\":{\".\":{},\"f:image\":{},\"f:imagePullPolicy\":{},\"f:name\":{},\"f:ports\":{\".\":{},\"k:{\\\"containerPort\\\":80,\\\"protocol\\\":\\\"TCP\\\"}\":{\".\":{},\"f:containerPort\":{},\"f:protocol\":{}}},\"f:resources\":{},\"f:terminationMessagePath\":{},\"f:terminationMessagePolicy\":{}}},\"f:dnsPolicy\":{},\"f:enableServiceLinks\":{},\"f:nodeSelector\":{},\"f:restartPolicy\":{},\"f:schedulerName\":{},\"f:securityContext\":{},\"f:terminationGracePeriodSeconds\":{},\"f:topologySpreadConstraints\":{\".\":{},\"k:{\\\"topologyKey\\\":\\\"kubernetes.io/hostname\\\",\\\"whenUnsatisfiable\\\":\\\"ScheduleAnyway\\\"}\":{\".\":{},\"f:labelSelector\":{},\"f:maxSkew\":{},\"f:topologyKey\":{},\"f:whenUnsatisfiable\":{}}}}} Subresource:} {Manager:kubelet.exe Operation:Update APIVersion:v1 Time:2023-09-27 10:25:27 -0700 PDT FieldsType:FieldsV1 FieldsV1:{\"f:status\":{\"f:conditions\":{\"k:{\\\"type\\\":\\\"ContainersReady\\\"}\":{\".\":{},\"f:lastProbeTime\":{},\"f:lastTransitionTime\":{},\"f:status\":{},\"f:type\":{}},\"k:{\\\"type\\\":\\\"Initialized\\\"}\":{\".\":{},\"f:lastProbeTime\":{},\"f:lastTransitionTime\":{},\"f:status\":{},\"f:type\":{}},\"k:{\\\"type\\\":\\\"Ready\\\"}\":{\".\":{},\"f:lastProbeTime\":{},\"f:lastTransitionTime\":{},\"f:status\":{},\"f:type\":{}}},\"f:containerStatuses\":{},\"f:hostIP\":{},\"f:phase\":{},\"f:podIP\":{},\"f:podIPs\":{\".\":{},\"k:{\\\"ip\\\":\\\"192.168.2.28\\\"}\":{\".\":{},\"f:ip\":{}}},\"f:startTime\":{}}} Subresource:status}]} Spec:{Volumes:[{Name:kube-api-access-kf7vs VolumeSource:{HostPath:nil EmptyDir:nil GCEPersistentDisk:nil AWSElasticBlockStore:nil GitRepo:nil Secret:nil NFS:nil ISCSI:nil Glusterfs:nil PersistentVolumeClaim:nil RBD:nil FlexVolume:nil Cinder:nil CephFS:nil Flocker:nil DownwardAPI:nil FC:nil AzureFile:nil ConfigMap:nil VsphereVolume:nil Quobyte:nil AzureDisk:nil PhotonPersistentDisk:nil Projected:&ProjectedVolumeSource{Sources:[]VolumeProjection{VolumeProjection{Secret:nil,DownwardAPI:nil,ConfigMap:nil,ServiceAccountToken:&ServiceAccountTokenProjection{Audience:,ExpirationSeconds:*3607,Path:token,},},VolumeProjection{Secret:nil,DownwardAPI:nil,ConfigMap:&ConfigMapProjection{LocalObjectReference:LocalObjectReference{Name:kube-root-ca.crt,},Items:[]KeyToPath{KeyToPath{Key:ca.crt,Path:ca.crt,Mode:nil,},},Optional:nil,},ServiceAccountToken:nil,},VolumeProjection{Secret:nil,DownwardAPI:&DownwardAPIProjection{Items:[]DownwardAPIVolumeFile{DownwardAPIVolumeFile{Path:namespace,FieldRef:&ObjectFieldSelector{APIVersion:v1,FieldPath:metadata.namespace,},ResourceFieldRef:nil,Mode:nil,},},},ConfigMap:nil,ServiceAccountToken:nil,},},DefaultMode:*420,} PortworxVolume:nil ScaleIO:nil StorageOS:nil CSI:nil Ephemeral:nil}}] InitContainers:[] Containers:[{Name:noop Image:mcr.microsoft.com/oss/kubernetes/pause:3.6 Command:[] Args:[] WorkingDir: Ports:[{Name: HostPort:0 ContainerPort:80 Protocol:TCP HostIP:}] EnvFrom:[] Env:[] Resources:{Limits:map[] Requests:map[] Claims:[]} ResizePolicy:[] RestartPolicy:<nil> VolumeMounts:[{Name:kube-api-access-kf7vs ReadOnly:true MountPath:/var/run/secrets/kubernetes.io/serviceaccount SubPath: MountPropagation:<nil> SubPathExpr:}] VolumeDevices:[] LivenessProbe:nil ReadinessProbe:nil StartupProbe:nil Lifecycle:nil TerminationM...+224150 more"
```

Reply:
Super verbose, fixed it to just print number of pods